pandas/tests/copy_view/test_indexing.py

```diff
@@ -159,10 +161,14 @@ def test_subset_column_slice(backend, using_copy_on_write, using_array_manager,
 
         subset.iloc[0, 0] = 0
         assert not np.shares_memory(get_array(subset, "b"), get_array(df, "b"))
-
     else:
         # we only get a warning in case of a single block
-        warn = SettingWithCopyWarning if single_block else None
+        # TODO
+        warn = (
+            SettingWithCopyWarning
```

Comment:
I am confused, shouldn't this be silent when CoW is enabled as well?

Reply:
Yes, and that `using_copy_on_Write` case is already handled in the `if` block above, so in this `else` I still need to handle the `warn_copy_on_write` case. 

I could actually do a `elif warn_copy_on_write` separate block, that might be easier to read and which is also the pattern I started to use in many other places.